File: 1000/2191B/main.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    int t;
    cin>>t;
    
    while(t--)
    {
        int n;
        cin>>n;
        
        unordered_map<int,int> mp;
        unordered_map<int,int> mp1;
        
        vector<int> a(n);
        for(int i=0;i<n;i++)
        {
            cin>>a[i];
            mp[a[i]]++;
        }
        
        sort(a.begin(), a.end());
        
        int l;
        int r;
        if(a[0] != 0)
            l = 0;
        else
            l = 1;
        
        bool flag = true;
        for(int i=0;i<n-1;i++)
        {
            mp[a[i]]--;
            mp1[a[i]]++;
            
            l = 0;
            while(mp1[l] > 0)
                l++;
            
            r = 0;
            while(mp[r] > 0)
                r++;
            
            if(l == r)
            {
                flag = false;
                break;
            }
        }
        
        if(flag)    
            cout<<"YES"<<endl;
        else
            cout<<"NO"<<endl;
    }
    
    return 0;
}

File: 1000/2191B/main.java
import java.util.*;

public class main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while (t-- > 0) 
        {
            int n = sc.nextInt();
            
            Map<Integer, Integer> mp = new HashMap<>();
            Map<Integer, Integer> mp1 = new HashMap<>();
            
            int[] a = new int[n];
            for (int i = 0; i < n; i++) 
            {
                a[i] = sc.nextInt();
                mp.put(a[i], mp.getOrDefault(a[i], 0) + 1);
            }
            
            Arrays.sort(a);
            
            int l;
            if (a[0] != 0)
                l = 0;
            else
                l = 1;
            
            boolean flag = true;
            for (int i = 0; i < n - 1; i++) 
            {
                // decrease count in mp
                mp.put(a[i], mp.get(a[i]) - 1);
                // increase count in mp1
                mp1.put(a[i], mp1.getOrDefault(a[i], 0) + 1);
                
                l = 0;
                while (mp1.getOrDefault(l, 0) > 0)
                    l++;
                
                int r = 0;
                while (mp.getOrDefault(r, 0) > 0)
                    r++;
                
                if (l == r) 
                {
                    flag = false;
                    break;
                }
            }
            
            if (flag)
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }
}
